lock-api/src/test/java/com/palantir/lock/client/MultiClientTransactionStarterTest.java

```diff
@@ -166,6 +169,8 @@ public void shouldNotFreeResourcesIfRequestIsServed() {
         verify(LOCK_CLEANUP_SERVICE_MAP.get(alpha), never()).unlock(any());
         verify(LOCK_CLEANUP_SERVICE_MAP.get(beta)).refreshLockLeases(any());
         verify(LOCK_CLEANUP_SERVICE_MAP.get(beta)).unlock(any());
+        verify(NAMESPACE_CACHE_MAP.get(alpha), never()).removeTransactionStateFromCache(anyLong());
+        verify(NAMESPACE_CACHE_MAP.get(beta), times(5)).removeTransactionStateFromCache(anyLong());
     }
```

Comment:
From above, the queue for beta here is [25] which means we cannot server any request. So the 5 responses we have from the first server call are transient and will be cleared.